### PR TITLE
fuzzing for wallet.ChooseSpends

### LINKED ISSUE #1251: Property & fuzz tests of the advanced spend API
Easier to test this at the wallet.Service layer

Use https://golang.org/pkg/testing/quick/ or some other property tester.

Also look into fuzzing it, if that could produce different results.

### PR DESCRIPTION
**Pull request by Vavilen**
Fixes #1251 

Changes:
- fuzzing for wallet.ChooseSpends

Does this change need to mentioned in CHANGELOG.md?

Tested methods:
- [x] ChooseSpends
- [x] CreateAndSignTransactionAdvanced

### CODE REVIEW

File: src/testutil/testutil.go [hunk 1]
```diff
@@ -43,6 +43,11 @@ func RequireError(t *testing.T, err error, msg string) {
 	require.Equal(t, msg, err.Error())
 }
 
+// EqualError checks that an error is not nil and that its message matches
+func EqualError(err error, msg string) bool {
```
Comment (gz-c) [inline src/testutil/testutil.go]: testutil.RequireError already does this

File: src/wallet/wallet_test.go [hunk 2]
```diff
@@ -1416,6 +1418,160 @@ func TestWalletChooseSpendsMinimizeUxOutsRandom(t *testing.T) {
...
+		return true
+	}
+	cfg := &quick.Config{
+		MaxCount: 1000000,
+		Rand:     rand.New(rand.NewSource(time.Now().Unix())),
+		Values: func(args []reflect.Value, rand *rand.Rand) {
+			uxb := makeRandomUxBalances(t)
+			args[0] = reflect.ValueOf(uxb)
+			args[1] = reflect.ValueOf(uint64(rand.Intn(100000)))
+			args[2] = reflect.ValueOf(uint64(0))
```
Comment (therealssj) [inline src/wallet/wallet_test.go]: why hours hardcoded to 0?

File: src/testutil/testutil.go (hunk 1 above)
Comment (gz-c) [reply src/testutil/testutil.go]: oh, just leave this method in the wallet_test.go file then

File: src/wallet/wallet_test.go [hunk 3]
```diff
@@ -1396,26 +1398,361 @@ func TestWalletChooseSpendsMaximizeUxOuts(t *testing.T) {
...
+		}
+
+		// Amongst the UxBalances with non-zero hours, they should be sorted as specified
+		ok = verifySortedCoinsFuzz(nonzeroBalances, cmpCoins)
+		if !ok {
+			log.Println("verifySortedCoinsFuzz(nonzeroBalances, cmpCoins) returned false")
+			return false
+		}
+
+		// If there are any extra UxBalances with non-zero hours, all of the zeros should have been chosen
```
Comment (iketheadore) [inline src/wallet/wallet_test.go]: This comment is outdated, not all zeros should have been chosen.

**Pull request closed without merging**